File: erpyoupal/clockify_accounting/report/clockify_cost_report/clockify_cost_report.py
# Copyright (c) 2013, Frappe Technologies Pvt. Ltd. and contributors
# For license information, please see license.txt

from frappe import _
import frappe
from frappe.utils import fmt_money, flt, getdate

def execute(filters=None):
	columns = get_columns(filters)
	data = get_data(filters)
	chart_data = get_chart_data(data)
	report_summary = get_report_summary(data)
	return columns, data, None, chart_data, report_summary

def get_columns(filters):
	columns = [
		{
			'label': _('Full name'),
			'fieldtype': 'Data',
			'fieldname': 'full_name',
			'width': 200
		},
		{
			'label': _('User'),
			'fieldtype': 'Data',
			'fieldname': 'user',
			'width': 200
		},
		{
			'label': _('Project'),
			'fieldtype': 'Data',
			'fieldname': 'project',
			'width': 200
		},
		{
			'label': _('Cost'),
			'fieldtype': 'Data',
			'fieldname': 'cost',
			'width': 200
		},
		{
			'label': _('Revenue'),
			'fieldtype': 'Data',
			'fieldname': 'revenue',
			'width': 200
		}
	]
	return columns

def get_data(filters):
	data = []

	return data

def get_chart_data(filters):
	chart_data = []

	return chart_data

def get_report_summary(filters):
	report_summary = []

	return report_summary

File: erpyoupal/clockify_accounting/report/clockify_cost_report/clockify_cost_report.js
// Copyright (c) 2016, Frappe Technologies Pvt. Ltd. and contributors
// For license information, please see license.txt
/* eslint-disable */

frappe.query_reports["Clockify Cost Report"] = {
	"filters": [
		{
			"fieldname": "month",
			"label": __("Month"),
			"fieldtype": "Select",
			"options": [
				{ "value": "January", "label": __("January") },
				{ "value": "February", "label": __("February") },
				{ "value": "March", "label": __("March") },
				{ "value": "April", "label": __("April") },
				{ "value": "May", "label": __("May") },
				{ "value": "June", "label": __("June") },
				{ "value": "July", "label": __("July") },
				{ "value": "August", "label": __("August") },
				{ "value": "September", "label": __("September") },
				{ "value": "October", "label": __("October") },
				{ "value": "November", "label": __("November") },
				{ "value": "December", "label": __("December") }
			],
			"default": "November",
		},
		{
			"fieldname": "workspace_id",
			"label": __("Workspace ID"),
			"fieldtype": "Link",
			"options": "Clockify Workspace",
			"default": "Youpal Group"
		},
		{
			"fieldname": "project_id",
			"label": __("Project ID"),
			"fieldtype": "Link",
			"options": "Clockify Projects",
			"get_query": function() {
				var workspace_id = frappe.query_report.get_filter_value('workspace_id');
				return{
					filters: {
						'workspace': workspace_id
					}
				};
			}
		},
	]
};
